### PR TITLE
Winter clean up 2016

### PR DESCRIPTION
**Pull request by Cadair**
This started as an attempt to fix various CI issues, but it escalated into me finding old bits of code lying about that needed cleaning. Currently contains:

* A switch from `astropy-helpers.sphinx.ext.numpydoc` to `sphinx.ext.napoleon` for NumPy style docstring parsing.
* Removal of the `Deprecated` decorator to favour `astropy.utils.decorator.deprecated`.
* Clean up of the GenericMap docstring.
* Removal of the `VSOClient.show` method which has been deprecated since v0.6.0.
* Removal of some placeholder docstrings as they were causing build issues.
* Change glymur to install from pip to fix the macos build.
* Update astropy_helpers to v1.3 to fix warnings on sphinx 1.5.
* Fixes to some database test numbers due to changes in the VSO results..
* Deprecate `sunpy.wcs`

### CODE REVIEW

**Comment by Cadair** [comment]
@sunpy/sunpy-developers review please, as this will fix the CU build.

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,9 @@ Latest
 * Remove the `sunpy.map.nddata_compat` module, this makes `Map.data` and
   `Map.meta` read only.
 * Add a `NorthOffsetFrame` class for generating HGS-like coordinate systems with a shifted north pole.
+* Remove deprecated `VSOClient.show` method.
+* Deprecate `sunpy.wcs`, `sunpy.coordinates` and `sunpy.map` now provide all
```
Comment (wafels) [inline CHANGELOG.md]: Tiny edit - replace the comma after `sunpy.wcs` with a colon to aid readability

**Comment by wafels** [comment]
Once this is accepted, I will adapt my remove_polyfun_at PR to take account of the changes to the util code.

**Comment by wafels** [comment]
I just made a PR to Cadair:travis-fix implementing the changes in the PR wafels:remove_polyfun_at .  Might as well roll them together.

**Comment by wafels** [comment]
+1

**Pull request merged**